File: src/main/scala/committee/nova/neutron/api/player/INeutronPlayer.scala
package committee.nova.neutron.api.player

import committee.nova.neutron.api.player.storage.{IHome, IPosWithDim}
import committee.nova.neutron.util.collection.LimitedLinkedList
import net.minecraft.entity.player.EntityPlayerMP

import java.util.{HashSet => JSet}

trait INeutronPlayer {
  def getRtpAccumulation: Int

  def setRtpAccumulation(acc: Int): Unit

  def getTpaCoolDown: Int

  def setTpaCoolDown(cd: Int): Unit

  def teleportTo(that: EntityPlayerMP): Boolean

  def getHomes: JSet[IHome]

  def setHomes(homes: JSet[IHome]): Unit

  def getFormerPos: LimitedLinkedList[IPosWithDim]

  def setFormerPos(former: LimitedLinkedList[IPosWithDim]): Unit
}

File: src/main/java/committee/nova/neutron/api/player/INeutronPlayer.java
package committee.nova.neutron.api.player;

import committee.nova.neutron.api.ITagSerializable;
import committee.nova.neutron.api.storage.INamedPos;
import committee.nova.neutron.api.storage.IPos;

import java.util.List;

public interface INeutronPlayer extends ITagSerializable {
    List<INamedPos> getHomes();

    List<IPos> getFootprints();

    boolean addHome(INamedPos home);

    boolean removeHome(String homeName);

    void addFootprint(IPos footprint);

    int getFootprintsLimit();
}
